ts/utils/wallet.ts

```diff
@@ -1,21 +1,13 @@
 import * as t from "io-ts";
 import { CreditCard } from "../types/pagopa";
+import { isExpired } from "./dates";
 import { NumberFromString } from "./number";
 /* 
     Contains utility functions to check conditions
     used across project (currently just in CardComponent)
  */
 
 export const isExpiredCard = (creditCard: CreditCard) => {
-  const today: Date = new Date();
```

Comment:
It seems ok!
Btw I noted the previous return statement is buggy (existing code, not yours)
`return ccExpireYear < cmpY || (ccExpireYear === cmpY && ccExpireMonth < cmpM);`
**_ccExpireYear_** (and **_ccExpireMonth_** as well) could be a number or a string.
I propose to use a default value for both of them so if we can't decode value we don't try to compare


```typescript
export const isExpiredCard = (creditCard: CreditCard) => {
  const decodedValueYear = NumberFromString.decode(creditCard.expireYear);
  const ccExpireYear = decodedValueYear.isRight()
    ? decodedValueYear.value
    : undefined;
  const decodedValueMonth = NumberFromString.decode(creditCard.expireMonth);
  const ccExpireMonth = decodedValueMonth.isRight()
    ? decodedValueMonth.value
    : undefined;
  // if we can't decode month or year value, card will be considered as expired
  if (ccExpireYear === undefined || ccExpireMonth === undefined) {
    return true;
  }
  return isExpired(ccExpireMonth, ccExpireYear);
};
```

With this we don't need to force a cast to _Number_
Could you try this?

Reply:
Ok, I can try this. But what about the force cast to Number in AddCardScreen?

Reply:
You can leave them. The cast is needed but we are sure about they are a numbers because before we made a value check with _**CreditCardExpirationMonth**_ and **_CreditCardExpirationYear_**

Reply:
Ok, done ;)